Examples/DicomConvert/DicomConvert.cxx

```diff
@@ -0,0 +1,181 @@
+ *  You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0.txt
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ *=========================================================================*/
+
+/**
+ * A SimpleITK example demonstrating how to convert and resize DICOM files
+ * to common image types.
+ */
+
+#include <SimpleITK.h>
+#include <iostream>
+#include <fstream>
+#include <vector>
+#include <filesystem>
+
+namespace sitk = itk::simple;
+namespace fs = std::filesystem;
+
+bool
+ConvertImage(const std::string & inputFile, const std::string & outputFile, unsigned int newWidth = 0)
+{
+  try
+  {
+    sitk::ImageFileReader reader;
+    // Only read DICOM images
+    reader.SetImageIO("GDCMImageIO");
+    reader.SetFileName(inputFile);
+    reader.ReadImageInformation();
+    auto                      iSize = reader.GetSize();
+    std::vector<unsigned int> imageSize;
+    imageSize.push_back(static_cast<unsigned int>(iSize[0]));
+    imageSize.push_back(static_cast<unsigned int>(iSize[1]));
+    // If 3D image with single slice, treat as 2D
+    if (iSize.size() == 3)
+    {
+      if (iSize[2] == 1)
+      {
+        imageSize.push_back(0U);
+      }
+      else
+      {
+        imageSize.push_back(static_cast<unsigned int>(iSize[2]));
+      }
+    }
+    reader.SetExtractSize(imageSize);
+    sitk::Image image = reader.Execute();
+    // Resample if new width is specified
+    if (newWidth > 0)
+    {
+      auto         originalSize = image.GetSize();
+      auto         originalSpacing = image.GetSpacing();
+      double       newSpacing = ((originalSize[0] - 1) * originalSpacing[0]) / (newWidth - 1);
+      unsigned int newHeight = static_cast<unsigned int>(((originalSize[1] - 1) * originalSpacing[1]) / newSpacing);
+      image = sitk::Resample(image,
+                             { newWidth, newHeight },
+                             sitk::Transform(),
+                             sitk::sitkLinear,
+                             image.GetOrigin(),
+                             { newSpacing, newSpacing },
+                             image.GetDirection(),
+                             0,
+                             image.GetPixelID());
+    }
+    // If a single channel image, rescale to [0,255]. Also modify the
+    // intensity values based on the photometric interpretation. If
+    // MONOCHROME2 (minimum should be displayed as black) we don't need to
+    // do anything, if image has MONOCHROME1 (minimum should be displayed as
+    // white) we flip the intensities. This is a constraint imposed by ITK
+    // which always assumes MONOCHROME2.
```

Comment:
So the goal here is that if the input in monochrome1 then to write the output as monochrome1? The important part to note here is that ITK inverts the intensity on read, but if output is monochrome1 then it does not automatically invert the intensity. (bug?)

Reply:
The script is converting the DICOM content to another file format which already requires intensity modifications from a high dynamic range to a low one. Because of the ITK convention of inverting MONOCHROME1 when reading, we need to invert it again so that when the file is saved it is visually similar to the original DICOM when displayed. I don't think this is a bug, it is the ITK convention and the desire to have consistent appearance of the converted image with the original DICOM content.

Reply:
I think I understand the logic less now.